bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java

```diff
@@ -246,49 +286,118 @@ protected String getRequestRequiredOptions() {
 
     @Override
     public StationProgram retrieveProgram(int station) throws CommunicationApiException {
-        JcResponse resp = statusInfo();
-        return resp.ps.stream().map(values -> new StationProgram(values.get(1))).collect(Collectors.toList())
-                .get(station);
+        if (jcReply.ps != null) {
+            return jcReply.ps.stream().map(values -> new StationProgram(values.get(1))).collect(Collectors.toList())
+                    .get(station);
+        }
+        return new StationProgram(0);
     }
 
-    private JcResponse statusInfo() throws CommunicationApiException {
+    private JcResponse statusInfo() throws CommunicationApiException, UnauthorizedApiException {
         String returnContent;
-
+        JcResponse resp;
         try {
             returnContent = http.sendHttpGet(getBaseUrl() + CMD_STATUS_INFO, getRequestRequiredOptions());
-        } catch (CommunicationApiException exp) {
+            resp = gson.fromJson(returnContent, JcResponse.class);
+            if (resp == null) {
+                throw new CommunicationApiException(
+                        "There was a problem in the HTTP communication: jcReply was empty.");
+            }
+        } catch (CommunicationApiException | JsonSyntaxException exp) {
             throw new CommunicationApiException(
                     "There was a problem in the HTTP communication with the OpenSprinkler API: " + exp.getMessage());
         }
-
-        JcResponse resp = gson.fromJson(returnContent, JcResponse.class);
         return resp;
     }
 
-    private static class JcResponse {
-        public List<List<Integer>> ps;
-        @SerializedName(value = "sn1", alternate = "rs")
-        public int rs;
-        public Integer curr;
+    private JoResponse getOptions() throws CommunicationApiException, UnauthorizedApiException {
+        String returnContent;
+        JoResponse resp;
+        try {
+            returnContent = http.sendHttpGet(getBaseUrl() + CMD_OPTIONS_INFO, getRequestRequiredOptions());
+            resp = gson.fromJson(returnContent, JoResponse.class);
+            if (resp == null) {
+                throw new CommunicationApiException(
+                        "There was a problem in the HTTP communication: joReply was empty.");
+            }
+        } catch (CommunicationApiException | JsonSyntaxException exp) {
+            throw new CommunicationApiException(
+                    "There was a problem in the HTTP communication with the OpenSprinkler API: " + exp.getMessage());
```

Comment:
Please include the cause exception in the new exception's constructor. That said it seems you are catching a CommunicationApiException and throwing yet another CommunicationApiException. Please consider either not catching the CommunicationApiException in the first place or rethrowing it.

Reply:
DONE: opted to not catch it in the first place as this means less code.